modules/openapi-generator/src/main/java/org/openapitools/codegen/config/CodegenConfigurator.java

```diff
@@ -643,6 +654,38 @@ public Context<?> toContext() {
         return new Context<>(specification, generatorSettings, workflowSettings);
     }
 
+    /*
+     * ODI-41 : Fix usage of x-protobuf-index for $ref fields
+     * This function automatically detect $ref that aren't part of an allOf and turn them into an allOf pattern
+     */
+    private void updateInputSpecReference() {
+        try {
+            String fileContent = new String(Files.readAllBytes(Paths.get(this.inputSpec)));
+            Pattern pattern = Pattern.compile("\\n\\s*(?<!-\\s?)\\$ref\\s?:.*\\n(?![\\s\\S]*components:)");
```

Comment:
Not mandatory cause it seems to be call one a run but usually,
if a regex has a unmodifiable pattern, it should be done as a class field
(compiling a pattern is time consuming)

Reply:
Done.